### PR TITLE
Display message if user joins non-PL section on My PL page

### PR DESCRIPTION
**Pull request by TurnerRiley**
Displays a message on the My PL page if the user joins a non-PL section and displays a message on the Teacher Homepage if the user joins a PL section.

### Joining section on the My PL page
For a non-PL section (shows new message):
join non pl

For PL section (shows regular message):
join_pl_success

### Joining section not on the My PL page (Teacher Dashboard as an example)
For a non-PL section (shows regular message):
non pl on dashboard

For PL section (shows new message):
Join pl on non pl

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-1831)

## Testing story
Local testing.

### CODE REVIEW

File: apps/src/templates/studioHomepages/JoinSectionNotifications.jsx [hunk 1]
```diff
@@ -46,18 +54,34 @@ JoinSectionNotifications.propTypes = {
...
+  sectionName,
+  showMessageForJoiningNonPl,
+}) => (
   <Notification
     type="success"
     notice={i18n.sectionsNotificationSuccess()}
-    details={i18n.sectionsNotificationJoinSuccess({sectionName})}
+    details={
+      showMessageForJoiningNonPl ? (
+        <SafeMarkdown
```
Comment (kelbyhawn) [inline apps/src/templates/studioHomepages/JoinSectionNotifications.jsx]: TIL!

**Review by kelbyhawn — APPROVED** [review]
This looks good to me, but someone else may want to take a look too just in case!

File: apps/i18n/common/en_us.json [hunk 2]
```diff
@@ -2384,6 +2384,7 @@
   "sectionsNotificationJoinFull": "We couldn't add you to section {sectionId} because there are already {sectionCapacity} students in it. Please contact your teacher for help.",
   "sectionsNotificationJoinNotFound": "Section {sectionId} doesn't exist.",
   "sectionsNotificationJoinSuccess": "You joined section {sectionName}.",
+  "sectionsNotificationJoinSuccessForNonPl": "You joined section {sectionName}. This is not a Professional Learning section so it will be visible on the [Teacher Dashboard]({teacherDashboardUrl}).",
```
Comment (bethanyaconnor) [inline apps/i18n/common/en_us.json]: @dmcavoy is "Teacher Dashboard" a phrase we have shown to teachers in the past? Or is there a better name for that page?

**Review by bethanyaconnor — COMMENTED** [review]
Might be out of scope, but do we want the reverse notification as well? ie if a user joins a PL section from the teacher homepage, should we direct them to the professional learning landing page?

File: apps/i18n/common/en_us.json (hunk 2 above)
Comment (dmcavoy) [reply apps/i18n/common/en_us.json]: I think we call this the Teacher Homepage in general.

Comment (TurnerRiley) [reply apps/i18n/common/en_us.json]: I'll update that!

**Comment by TurnerRiley** [comment]
> Might be out of scope, but do we want the reverse notification as well? ie if a user joins a PL section from the teacher homepage, should we direct them to the professional learning landing page?

I think that info is covered by what's shown on the Teacher Dashboard from [this PR](https://github.com/code-dot-org/code-dot-org/pull/58102) but I'm open to making one for that too! @dmcavoy thoughts?

**Comment by bethanyaconnor** [comment]
hm I don't see what I'm thinking of. I'm thinking of text like "You joined section {sectionName}. This is a professional learning section and will be shown on the My Professional Learning page." (slightly different than the "your sections have moved" message as that's temporary)

**Comment by TurnerRiley** [comment]
> hm I don't see what I'm thinking of. I'm thinking of text like "You joined section {sectionName}. This is a professional learning section and will be shown on the My Professional Learning page." (slightly different than the "your sections have moved" message as that's temporary)

Sounds good! On it!

**Review by bethanyaconnor — APPROVED** [review]
🎉

**Pull request merged**